metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py

```diff
@@ -357,7 +366,11 @@ def _get_operation_aspect_work_unit(
             )
             reported_time: int = int(time.time() * 1000)
             last_updated_timestamp: int = int(start_time.timestamp() * 1000)
-            user_urn = make_user_urn(self.get_user_identifier(user_name, user_email))
+            user_urn = make_user_urn(
+                self.get_user_identifier(
+                    user_name, user_email, self.config.email_as_user_identifier
```

Comment:
Should we add logic to generate an email from `self.config.email_domain` if it's set, like we do in `_map_user_counts`? Perhaps it'd be best to put that logic in `get_user_identifier` instead

Reply:
let's keep it as is. The current method uses the user_email to determine whether to skip the current metadata (i.e continue in for loop)